cpmpy/solvers/minizinc.py

```diff
@@ -110,22 +112,68 @@ def __init__(self, cpm_model=None, solver=None):
             solvername = solvername[9:]
         self.name = "minizinc:"+solvername
 
-        import minizinc
+        # Create solver specific objects
         self.mzn_solver = minizinc.Solver.lookup(solvername)
-
-        # minizinc-python API
-        # Create a MiniZinc model
         self.mzn_model = minizinc.Model()
-        if cpm_model is None:
-            self.user_vars = []
-            self.mzn_txt_solve = "solve satisfy;"
-        else:
-            # store original vars and objective (before flattening)
-            self.user_vars = get_variables_model(cpm_model)
-            (mzn_txt, self.mzn_txt_solve) = self.make_model(cpm_model)
-            self.mzn_model.add_string(mzn_txt)
-            # do NOT add self.mzn_txt_solve yet, so that it can be overwritten later
+        self.mzn_model.add_string(CPM_minizinc.header)
+
+        #Solver text
+        self.mzn_txt_solve = "solve satisfy;"
+
+        super().__init__(cpm_model, solver=solver, name=solvername)
+
+    def __add__(self, cons):
+        """
+            Add an additional (list of) constraints to the model
+        """
+        # Add variables in cons
+        for var in get_variables(cons):
+            if var in self.user_vars:  # Might be possible to do this prettier?
+                continue
+            if isinstance(var, _BoolVarImpl):
+                self.mzn_model.add_string(f"var bool: {self.clean_varname(var.name)};\n")
+            elif isinstance(var, _IntVarImpl):
+                self.mzn_model.add_string(f"var {var.lb}..{var.ub}: {self.clean_varname(var.name)};\n")
+            self.user_vars.add(var)
+
+        if not is_any_list(cons):
+            cons = [cons]
+
+        # we can't unpack lists in convert_expression, so must do it upfront
+        # and can't make assumptions on '.flat' existing either...
+        # this is dirty code that should not be reused, keeping it hidden in this function...
+        # TODO: can't we put it in convert_expression now?
+        def flatlist(lst):
```

Comment:
expressions.utils now had a flatlist function, that one can probably be used instead so this can go